src/test/java/com/ge/predix/metering/filter/NuregoIT.java

```diff
@@ -16,125 +16,277 @@
+			} else {
+				componentId = retrieveComponent(uaaInstanceId, nuregoToken);
+			}
+			if (componentId == null) {
+				throw new IllegalStateException("Component ID is null because Nurego mapping is not updated.");
+			}
+		    this.meteringFilter.doFilter(request, response, new MockFilterChain());
+		    Thread.sleep(4000);
+		    this.meteringFilter.doFilter(request, response, new MockFilterChain());
+		    Thread.sleep(10000);
+	
+		    ComponentUsage usages = retrieveRawUsage(nuregoToken, componentId, acsFlag, featureId);
+		    Assert.assertEquals(usages.getCompCount(), 2);
+		    Assert.assertEquals(usages.getUsage(), 2.0);
+    }
+   
+    @AfterTest
+    private void cleanup() {
+        serviceInstanceHelper.deleteServiceInstance(acsInstanceId);
+        serviceInstanceHelper.deleteServiceInstance(uaaInstanceId);
+    }
+    private String getNuregoAuthToken() throws Exception{
+    	
+    		HashMap<String, Object> serviceRequest = new HashMap<String, Object>();
+		serviceRequest.put(Constants.USERNAME, this.nuregoUsername);
+		serviceRequest.put(Constants.PASSWORD, this.nuregoPassword);
+		serviceRequest.put(Constants.INSTANCE_ID, this.nuregoInstanceId);
 
-        return null;
+		URI oauthURL = URI.create(this.nuregoURL + Constants.NUREGO_AUTH_TOKEN_URL);
+		String response = this.nuregoTemplate.postForObject(oauthURL, serviceRequest, String.class);
+		@SuppressWarnings("unchecked")
+		Map<String, Object> responseMap = new ObjectMapper().readValue(response, Map.class);
+		return responseMap.get(Constants.ACCESS_TOKEN).toString();
     }
+    
+	private String retrieveComponent(String serviceInstanceGuid, String nuregoAccessToken) throws Exception{
+
+		String retrieveComponentURL = this.nuregoURL + Constants.SERVICE_COMPONENT_URL + serviceInstanceGuid;
+		
+		HttpHeaders headers = new HttpHeaders();
+		headers.add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+		headers.add(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+		headers.add(HttpHeaders.AUTHORIZATION, Constants.BEARER + nuregoAccessToken);
+		
+		UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(retrieveComponentURL)
+				.queryParam("provider", "cloud-foundry");
+		
+		HttpEntity<?> entity = new HttpEntity<>(headers);
+		ResponseEntity<String> responseEntity = null;
+		long end = System.currentTimeMillis() + 40000;
+		//Attempts to retrieve component for 20 seconds before it throws an exception.
+		//Component id is not updated immediately in Nurego after service instance creation, there is a delay. Hence trying it for 20 seconds.
+	    while(System.currentTimeMillis() < end) {
+	    		try {
+	    			responseEntity = this.nuregoTemplate.exchange(builder.build().encode().toUri(), HttpMethod.GET, entity ,String.class);
+	    			Thread.sleep(2000); // need to sleep because nurego mapping takes time to update
+	    			break;
+	    		} catch(Exception ex) {
+	    			System.out.println("Attempting to retrieve component ID...");
```

Comment:
This should probably be printed using a logger, even though it doesn't matter as much since it is a test.